gobblin-service/src/main/java/org/apache/gobblin/service/modules/template_catalog/FSFlowTemplateCatalog.java

```diff
@@ -125,13 +123,7 @@ public FlowTemplate getFlowTemplate(URI flowTemplateDirURI)
       return false;
     };
 
-    if (!this.sysConfig.hasPath(ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY)) {
-      throw new RuntimeException("Missing config " + ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY);
-    }
-    if (!flowTemplateDirURI.getScheme().equals(FS_SCHEME)) {
-      throw new RuntimeException(
-          "Expected scheme " + FS_SCHEME + " got unsupported scheme " + flowTemplateDirURI.getScheme());
-    }
+    validateTemplateURI(flowTemplateDirURI);
```

Comment:
Throw an exception if flowTemplateDirUri is invalid. Change it to if (!isValidTemplateURI()) {...}.